### PR TITLE
chore(docs): add how-to for rotating secrets

### PR DESCRIPTION
**Pull request by biancadanforth**
Relates to: [FXA-8371](https://mozilla-hub.atlassian.net/browse/FXA-8371)

### CODE REVIEW

File: docs/how-tos/rotating-secrets.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+## Confirm the leak and its severity
+
+Sometimes, sensitive looking data isn't actually valid or compromising. Try to answer the following questions, answered below with our Stripe API key example:
+
+* **Is this a valid secret?**
+:::note
+If the secret is expired or otherwise invalid, the urgency goes away, and you can skip to the next step. If the key is valid but only in sandbox/test environments with fake data, the urgency also goes away, but you'll still want to answer the questions below.
+:::
+  * Yes. It is listed in the [stage (`SUB_PLAT_STAGE`) dashboard](https://dashboard.stripe.com/test/apikeys) as the unrestricted [test-mode](https://stripe.com/docs/test-mode) secret key, and it is active. This is a real leak.
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:30]: Should "test-mode" be "test mode"?

Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:30]: Also wondering if you should swap the order of these two. I understand what you are doing and in one sense it is accurate to skip the extra effort in reading something that is inapplicable, but the note applies to all questions below. Should it maybe switched to above (reworded slightly) or swapped to improve reading flow and applicability?

File: docs/how-tos/rotating-secrets.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+* **Is it in use?**
+  * In this case, the [Stripe API Keys page in the dashboard](https://dashboard.stripe.com/test/apikeys) has helpful information including when a given valid key was last used. Our example key was last used earlier today, so it's definitely still in use!
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:32]: > Our example key was last used earlier today, so it's definitely still in use!

While this is accurate for right now, will this always be true? Maybe reword to say as of this week or something similar? I am hesitant of anything that mentions "today" as it is easily missed in updating in future and might not apply then (for whatever reason).

File: docs/how-tos/rotating-secrets.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+* **What do we use it for?**
+  * Since this is an unrestricted Stripe API key (test mode), it is used to make read/write API requests to any Stripe API in test mode in our staging environment. Without [searching for this key value explicitly in various places](#places-to-check-where-a-secret-might-be-used), one can only guess at this point that it's likely used by our stage apps to make Stripe API calls, and maybe by our CI end-to-end tests to do the same.
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:34]: This might be personal but from a SR perspective, I didn't relate this link:

> [searching for this key value explicitly in various places](#places-to-check-where-a-secret-might-be-used)

to this anchor:

> Places to check where a secret might be used

Is there perhaps a better way to make this obvious? Completely nit so not holding back merging, but maybe leave a note to fix in future?

File: docs/how-tos/rotating-secrets.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  * In terms of severity, the fact that this key is unrestricted is very bad, but the fact that it's only valid for our staging environment (and in Stripe test mode), which uses fake data and money, greatly reduces the risk and urgency of this leak. 
+    * By contrast, if this same key was for production (live mode), an attacker could [make charges or issue refunds](https://stripe.com/docs/keys#safe-keys) among other things.
+  * **Can someone with this secret access sensitive user data?**
+    * No. All data is test data.
+  * **Can someone with this secret modify critical user or application state?**
+    * They could not modify real users' state. They could modify our stage application state and/or cause us to get rate limited by Stripe, where subsequent API requests are denied.
+* **Where was it exposed?**
+  * The GitHub secret scanning alert indicates where it found the secret referenced. It scans the code but also other forms of user-generated content, like PR comments and issues. It's worth noting that depending on how old the alert is, a reference may have since been removed or moved.
+  * In our case, this was leaked in a user comment within a GitHub issue.
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:44]: Where was this referenced?

File: docs/how-tos/rotating-secrets.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+In this example, the secret was leaked in a GitHub issue comment. The comment can be edited to redact the reference, but it will still be retained in the comment's revision history. An [admin for the repo](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization) can remove the revision which introduced the leak to completely remove any public reference to it.
+
+## Rotate the secret (if valid)
+
+Now that we've stopped the bleeding, it's time to rotate the secret if it was determined to be valid and in use. 
+
+The steps here are to:
+1. create a new secret,
+2. replace the old secret where it's used with the new secret, and
+3. invalidate the old secret
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:60]: Nit: not sure of grammar here, but unordered bullets are capitalized and numbers here are not. Also, semicolon instead of comma, after 'secret' in (2).

File: docs/how-tos/rotating-secrets.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+Depending on the urgency and deployment process, this overlap period where both keys are valid could be on the order of seconds, hours or even days. Discuss this with the team if you're uncertain.
+:::
+
+For our example, [Stripe provides some helpful docs](https://stripe.com/docs/keys#rolling-keys) on how to create a new secret and invalidate the old secret. The hard part here is replacing the old secret where it is used with the new secret, as we store secrets in a number of different places for different purposes.
+
+### Places to check where a secret might be used
+* Application config environment variables
+  * These secrets are managed by SRE, so you will need to [securely send them the new secret](https://mozilla-hub.atlassian.net/wiki/spaces/SECENGOPS/pages/378273870/How+to+share+a+secret+with+coworkers), so they can update it and redeploy the affected application(s).
+  * If you're unsure what environment variable(s) represent this secret in what app(s), you can send them both the old secret and the new secret, so they can do a find-and-replace.
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:73]: 👍 Excellent note!

File: docs/how-tos/rotating-secrets.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+* [CircleCI environment variables](https://circleci.com/docs/env-vars/#private-keys-and-secrets)
+* GitHub [Actions](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions), [Codespaces](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-secrets-for-your-codespaces) and/or [Dependabot](https://docs.github.com/en/code-security/dependabot/working-with-dependabot) environment variables
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:75]: See above comment. Same here! 👍

File: docs/how-tos/rotating-secrets.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+:::caution
+This is not an exhaustive list. Check with other engineers and/or SRE if you are unsure where all a secret may be kept.
```
Comment (sardesam) [inline docs/how-tos/rotating-secrets.md:78]: Nit: reword. Grammar appears a bit off here:

> if you are unsure where all a secret may be kept.

File: docs/how-tos/rotating-secrets.md (hunk 1 above)
Comment (biancadanforth) [reply docs/how-tos/rotating-secrets.md:30]: Sure I'll switch it to above.

File: docs/how-tos/rotating-secrets.md (hunk 2 above)
Comment (biancadanforth) [reply docs/how-tos/rotating-secrets.md:32]: How about instead of "today", I say "earlier in the day"?

Comment (biancadanforth) [reply docs/how-tos/rotating-secrets.md:32]: Discussed sync with Meg; we'll just say: "it was used very recently".

File: docs/how-tos/rotating-secrets.md (hunk 8 above)
Comment (biancadanforth) [reply docs/how-tos/rotating-secrets.md:78]: Reworded to "if you are unsure of where all of the locations where a secret may be kept".

**Review by sardesam — APPROVED** [review]
r+ !!!
